src/interface/variable.hpp

```diff
@@ -54,6 +54,20 @@ inline std::string MakeVarLabel(const std::string &base_name, int sparse_id) {
          (sparse_id == InvalidSparseID ? "" : "_" + std::to_string(sparse_id));
 }
 
+struct VariableState : public empty_state_t {
+  KOKKOS_INLINE_FUNCTION
+  VariableState(double alloc, double dealloc)
+      : sparse_allocation_threshold(alloc), sparse_deallocation_threshold(dealloc) {}
+
+  KOKKOS_DEFAULTED_FUNCTION
+  VariableState() = default;
+  KOKKOS_INLINE_FUNCTION
+  explicit VariableState(const empty_state_t &) {}
+
+  double sparse_allocation_threshold;
+  double sparse_deallocation_threshold;
```

Comment:
Should these `threshold`s be `const`?

Reply:
When I made them `const` there were some issues with copying the ParArrays around.

Actually, this code probably should not be in this pull request since I am not actually changing how the arrays in `CellVariable` behave. I am going to remove it for now.